### DISCUSSION TITLE
How to index non-extension or wrong extension file ?

### DISCUSSION THREAD

**Discussion by Alceatraz** [photoprism/photoprism#4225] [Q&A]
Hello!

Here is what I'm doing: I want index (not import) all picture from my [OSS](https://github.com/Alceatraz/lite-oss). All uploaded file rename into it's sha256 like:

> 334c3fd7b84....[total 64 char long]....e49ce81d4e883f5
> d43c5ddcabe....[total 64 char long]....a08c0f709472859

Seems like photoprism will ignore no extension named file and skip wrong extension file. I have 12600 pictures:

1. only file named like `334c3fd`
> INFO index: updated 0 file

2. File with wrong extension xxx.gif with a png file
> ERRO index: skipped hash.gif due to invalid extension for media type 'image/png'

FYI:
- I ran photoprism in truenas scale
- I set photoprism Read-Only Mode
- OSS server write file into originals

Question: How to index all file and auto detcet media type?

**Comment by lastzero** [comment]
Having a proper media file extension is essential, also for compatibility with the tools that PhotoPrism uses in the background to process your files. You can, for example, run a batch script similar to this one to detect the file type and add the corresponding extension:
https://github.com/photoprism/photoprism-contrib/blob/main/scripts/Batch%20Rename/heic2jpg.sh

**Reply by Alceatraz** [reply]
Hello! Thanks for reply

But here is the problem: Because of object storage service use hash to identify file, Rename them will cause "No such file" which mean all data will lost. So rename any file is unacceptable.

Is that a switchable validtion function OR total relay on media file extension?

**Reply by lastzero** [reply]
You could use symbolic links in that case.
